api/http/handler/webhooks/webhook_create.go

```diff
@@ -0,0 +1,66 @@
+package webhooks
+
+//TODO
+// * Generate random string for TokenData for added security
+
+import (
+	"net/http"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/portainer/portainer"
+	httperror "github.com/portainer/portainer/http/error"
+	"github.com/portainer/portainer/http/request"
+	"github.com/portainer/portainer/http/response"
+)
+
+var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890")
+
+// Used for generating random token
+
+type webhookCreatePayload struct {
+	ServiceID  string
+	EndpointID int
+}
+
+func (payload *webhookCreatePayload) Validate(r *http.Request) error {
+	if govalidator.IsNull(payload.ServiceID) {
+		return portainer.Error("Missing ServiceID")
```

Comment:
To keep consistent with other validation messages, please use the form `Invalid ServiceID`.